File: libs/grpc/clients/parser.go
package clients

import (
	"log"

	"github.com/twirapp/twir/libs/grpc/constants"
	"github.com/twirapp/twir/libs/grpc/parser"
	"google.golang.org/grpc"
)

func NewParser(env string) parser.ParserClient {
	serverAddress := createClientAddr(env, "parser", constants.PARSER_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		defaultClientsOptions...,
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := parser.NewParserClient(conn)
	return c
}

File: libs/grpc/clients/parser.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc'

import { CLIENT_OPTIONS, createClientAddr, waitReady } from './helper'
import { PORTS } from '../constants/constants.js'
import { ParserDefinition } from '../parser/parser'

import type { ParserClient } from '../parser/parser'

export async function createParser(env: string): Promise<ParserClient> {
	const channel = createChannel(
		createClientAddr(env, 'parser', PORTS.PARSER_SERVER_PORT),
		ChannelCredentials.createInsecure(),
		CLIENT_OPTIONS,
	)

	await waitReady(channel)

	const client = createClient(ParserDefinition, channel)

	return client as any
}
